### ISSUE TITLE
WebCodecs Time Stamps

### ISSUE THREAD

**Issue by sandersdan** [w3c/hr-time#104]
Hello!

We're trying to work out (https://github.com/WICG/web-codecs/issues/52, https://github.com/WICG/web-codecs/issues/122) what kind of timestamps to use for frames in WebCodecs. In general we need ~microsecond resolution, measured from either time 0 at the start of the media, or in calendar time (typically relative to the Unix epoch).

Usually media times are measured as an integer fraction of seconds (a timestamp and a timebase), but there doesn't seem to be a convenient Rational type in JS, so it's not very ergonomic.

The &lt;video&gt; element has `currentTime` in seconds (as a double). RTC uses DOMHighResTimeStamp for some things. Internally Chrome uses int64 microseconds.

It seems like DOMHighResTimeStamp is the most reasonable choice for us, but it also seems to have been defined relative to a specific epoch. Is there any concern with WebCodecs using it with no explicit epoch?

**Comment by npm1** [comment]
You can use DOMHighResTimeStamp , which is just a typedef for double. From the spec: `The DOMHighResTimeStamp type is used to store a time value in milliseconds, measured relative from the time origin, global monotonic clock, or a time value that represents a duration between two DOMHighResTimeStamps.` So it should be in milliseconds (not seconds), and it can represent any time duration (a duration is not relative to anything).

**Comment by sandersdan** [comment]
Thanks!

**Comment by mdrejhon** [comment]
I agree with microsecond timestamps -- **not milliseconds**

For futureproofing testing, we did some tests of 240fps HFR videos and 360fps HFR videos on gaming 240Hz and 360Hz monitors.  ASUS has roadmapped a 1000Hz gaming monitor as well.  In some browsers the ultra-high-framerate videos work fine, but some issues like synchronization and stutters occur.  

Some useful reading may be warranted: [Ultra HFR: Real-Time 240fps, 480fps and 1000fps Videos](https://www.blurbusters.com/faq/ultrahfr)

We successfully tested Google Chrome to 480fps on an experimental 480Hz monitor, www.blurbusters.com/480hz

For those who want to read more about recent scientific research on refresh rate physics, there is an Area51 section at Blur Busters that is being worked with display engineers -- www.blurbusters.com/area51

To futureproof adequately, we strongly believe in microsecond timestamps, considering the human-visible benefits of the millisecond (discussed in this forum thread https://forums.blurbusters.com/viewtopic.php?f=7&t=6808 ...)
